### PR TITLE
Compliance tests for failures

### PR DESCRIPTION
**Pull request by alekcz**
- Ensure all protocol methods throw errors 
- Ensure all protocol methods throw their errors using the async channel

### CODE REVIEW

File: src/konserve/compliance_test.cljc [hunk 1]
```diff
@@ -50,6 +59,27 @@
         (every?
          (fn [{:keys [:konserve.core/timestamp]}]
            (= (type (java.util.Date.)) (type timestamp)))
-         list-keys)))))
+         list-keys)))
+    
+    (let [corrupt (-> store ; let's corrupt our store
+                    (clojure.core/assoc-in [:config] (UnknownType.)) ;common naming convention
+                    (clojure.core/assoc-in [:con] (UnknownType.))
```
[the remark this answers is not in the document]
Comment (alekcz) [reply src/konserve/compliance_test.cljc]: @whilo I've updated it to dynamically replace all the properties of the store

**Comment by whilo** [comment]
I somewhat get the sense for this kind of testing, but I am still not sure whether this corruption should not be stochastic and somewhat deeper in the logic/datastructure. At least right now we are not necessarily testing the full error propagation. I am still fine with merging this though, it cannot harm.

**Comment by alekcz** [comment]
The main idea of the corruption is to ensure that interactions with the underlying store always fails. In doing so the we can make sure that the backend handles these errors gracefully while propagating them in adherence with the protocol. 

FYI: When I last checked the filestore failed these compliance tests.

**Review by whilo — APPROVED** [review]
Nice work!

**Pull request merged**